types/angular/index.d.ts

```diff
@@ -1520,23 +1520,27 @@ declare namespace angular {
         (data: T, status: number, headers: IHttpHeadersGetter, config: IRequestConfig): void;
     }
 
-    interface IHttpPromiseCallbackArg<T> {
-        data?: T;
-        status?: number;
-        headers?: IHttpHeadersGetter;
-        config?: IRequestConfig;
-        statusText?: string;
+    interface IHttpResponse<T = any> {
+        data: T;
+        status: number;
+        headers: IHttpHeadersGetter;
+        config: IRequestConfig;
```

Comment:
I like this change. I never found out why they were set to optional to begin with. They always seemed to be set. This will let people get rid of a lot of non-null assertion operators.